src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/service/CourtCaseEventsService.kt

```diff
@@ -49,9 +47,9 @@ class CourtCaseEventsService(
             createNewPersonRecordAndProcess(person, pncId)
           }
         }
-      } else {
-        trackEvent(INVALID_PNC, mapOf("PNC" to pncIdentifier.toString()))
       }
+      is InvalidPNCIdentifier -> trackEvent(INVALID_PNC, mapOf("PNC" to pncIdentifier.invalidValue()))
+      else -> trackEvent(MISSING_PNC, emptyMap())
     }
```

Comment:
A small suggestion if you are interested! we could extract further so that the code will looks like this

`    when (val pncIdentifier = person.otherIdentifiers?.pncIdentifier) {
      is ValidPNCIdentifier -> processValidPNCRecord(pncIdentifier, person)
      is InvalidPNCIdentifier -> processInvalidPNCRecord(pncIdentifier)
      else -> trackEvent(MISSING_PNC, emptyMap())
    }`